zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/job/JobBatchActivateProcessor.java

```diff
@@ -122,8 +130,8 @@ private void rejectCommand(final TypedRecord<JobBatchRecord> record) {
       rejectionType = RejectionType.INVALID_ARGUMENT;
       rejectionReason = String.format(format, "type", "present", "blank");
     } else {
-      throw new IllegalStateException(
-          "Expected to reject an invalid activate job batch command, but it appears to be valid");
+      rejectionType = RejectionType.NOT_FOUND;
+      rejectionReason = "No jobs found to activate";
```

Comment:
❌ Before in gRPC we would return an error with the message
`"Expected to handle gRPC request ActivateJobs with tenant identifiers '%s', but tenant is not authorized to perform this request";`

This would have a `PERMISSION_DENIED_VALUE` error type.

We should try to keep this aligned as much as we can. So I'd propose if the user is not authorized for all the tenants provided in the request we return an `UNAUTHORIZED` rejection type with an error message saying that the user is not provided to activate jobs for tenant X.

Reply:
We also need to add a test case for this